packages/react-form/src/hooks/field/reducer.ts

```diff
@@ -138,5 +150,6 @@ export function initialFieldState<Value>(value: Value): FieldState<Value> {
     error: undefined,
     touched: false,
     dirty: false,
+    allErrors: [],
```

Comment:
Because, `allErrors` is a non-primitive type, it cannot be consumed in the same way as `dirty`, or `touched` or other primitive values.  What is the probability of at least one consumer using the `allErrors` array like this:

```tsx
React.useEffect(() => {
  /* AllErrors is assumed to have changed. Do something imperative, like a Toast message */
}, [field.allErrors]
```

My concern is that it could create a really poor DX.  Developers would expect to be able to detect meaningful changes to allErrors, but it will change all the time, as new array instances are created even if the contents of the array aren't meaningfully different.

I also understand there would be some real reluctance to make `useField` depend on an immutability library, as that would affect a large percentage of bundle sizes.

Reply:
I intended to click "Comment", not "Request Changes".   Please feel free to dismiss my review if it's not relevant.

Reply:
@jared-dykstra I see your point. Do you have any suggestion for a change that wouldn't have such an impact of developer experience ?

Reply:
I think it's a bit tricky to solve.  Some ideas:
1. Make the reducer immutable, using immer.js.
   * This would be a rather significant change due to deps and bundle size implications, so there is plenty of reason to *not* go this route
   * However, more and more libraries are embracing immutable data
1. Stick with primitive types
   * `allErrors` would have to be exposed a different way, perhaps as the return value of `runValidation()`  However, that would provide a Developer Experience which encourages imperative programming.  It wouldn't mesh very will with React's Functional Programming paradigm.
1. Carefully manage the array instance:
   1. Create a function to update the previous value of `allErrors` by comparing each array element: If everything is the same, re-use the previous instance.  If anything has changed, create a new array instance.

IMO, the best option is to make the state immutable.  The 2nd best option is to carefully manage the array instance.

I'd recommend adding at least one test: